CONTRIBUTING.md

```diff
@@ -1,303 +1,303 @@
+export GOPATH=$HOME/go
+mkdir -p $GOPATH/src/github.com/goharbor
+
+#Get code
+go get github.com/goharbor/harbor
+cd $GOPATH/src/github.com/goharbor/harbor
+
+#Track repository under your personal account
+git config push.default nothing # Anything to avoid pushing to goharbor/harbor by default
+git remote rename origin goharbor
+git remote add $USER git@github.com:$USER/harbor.git
+git fetch $USER
+
+```
+**NOTES:** Note that GOPATH can be any directory, the example above uses $HOME/go. Change $USER above to your own GitHub username.
+
+To build the project, please refer the [build](docs/compile_guide.md) guideline.
+
+### Repository Structure
+
+Here is the basic structure of the harbor code base. Some of the key folders / files are commented for your references.
+```
+.
+...
+├── Makefile      # Make file for compile and build code
+├── contrib       # Contain documents, scripts, and other helpful things which are contributed by the community
+├── docs          # Keep documents here
+├── make          # Resource for building and setting up Harbor environment
+...
+├── src           # Source code folder
+├── tests         # Test cases for API / e2e testings
+└── tools         # Keep supporting tools
+...
+```
+
+The folder graph below shows the structure of the source code folder `harbor/src`, which will be your primary working directory. The key folders are also commented.
+```
+.
+├── adminserver          # Source code for the admin server component
+│   ├── api
+│   ├── auth
+│   ├── client
+│   ├── handlers
+│   ├── systemcfg
+│   └── systeminfo
+├── common              # Source code for some general components like dao etc.
+│   ├── api
+│   ├── config
+│   ├── dao
+│   ├── models
+│   ├── notifier
+│   ├── scheduler
+│   ├── secret
+│   ├── security
+│   └── utils
+├── jobservice          # Source code for the job service component
+│   ├── api
+│   ├── config
+│   ├── job
+│   ├── replication
+│   ├── scan
+│   └── utils
+├── ui                  # Source code for the harbor service component
+│   ├── api
+│   ├── auth
+│   ├── config
+│   ├── controllers
+│   ├── filter
+│   ├── promgr
+│   ├── proxy
+│   ├── service
+│   ├── static
+│   ├── utils
+│   └── views
+├── portal               # The code of harbor web UI
+│   ├── e2e
+│   ├── lib             # Source code of @harbor/ui npm library which includes the main UI components of web UI
+│   └── src             # General web page UI code of Harbor
+└── vendor              # Go code dependencies
+    ├── github.com
+    ├── golang.org
+    ├── google.golang.org
+    └── gopkg.in
+```
+
+### Setup Development Environment
+
+#### Go
+Harbor backend is written in [Go](http://golang.org/). If you don't have a Harbor backend service development environment, please [set one up](https://golang.org/doc/install).
+
+|  Harbor  |  Requires Go  |
+|----------|---------------|
+|   1.1    |    1.7.3      |
+|   1.2    |    1.7.3      |
+|   1.3    |    1.9.2      |
+|   1.4    |    1.9.2      |
```

Comment:
How about appending the latest two other versions 1.5, 1.6 here?

Reply:
Do you happen to know what versions of Go were required for these two versions?

Reply:
@clouderati Same with 1.3&1.4, still keep at 1.9.2